File: CODE/RetrieveBoxscoresTable.py
# FILE: RetrieveBoxscoresTable.py
# DATE 2025-10-19
# DESCRIPTION: 
# This Python program retrieves all data from a specific table within an 
# SQLite database and displays it as a formatted table using the pandas 
# library. 

import sqlite3
import pandas as pd
import os

# Connect to database
path_to_database = os.path.join(os.getenv("EXAMPLES"), "Boxscores.db")

# Use a context manager (the 'with' statement) to handle the connection.
# This ensures the connection is automatically closed, even if errors occur.
try:
    with sqlite3.connect(path_to_database) as con:
        # The pandas read_sql function executes the SQL query and 
        # returns a DataFrame.
        query_text = "SELECT * FROM Boxscores"
        print(f"Executing query: {query_text}")

        results_df = pd.read_sql(query_text, con)

    print('-------------------------')
    print(results_df)
    print("Done")

except sqlite3.Error as e:
    print(f"An error occurred: {e}")

File: CODE/RetrieveBoxscoresTable.R
# Copyright 2025 by Steven J. Keyes. All rights reserved.
# FILE: Example_2_RetrieveBoxscoresTable.R
# DATE 2025-10-13
# DESCRIPTION: 

library(RSQLite)
library(DBI)
library(dplyr)

# Connect to database with error handling
path_to_database <- file.path(Sys.getenv("EXAMPLES"), "Boxscores.db")

tryCatch({
  # Check if file exists first
  if (!file.exists(path_to_database)) {
    stop(sprintf("Database file not found at: %s", path_to_database))
  }
  
  con <- dbConnect(RSQLite::SQLite(), path_to_database)
  
  # Get all data from Boxscores table
  query <- tbl(con, "Boxscores") |>
    select(everything())
  
  # Look at SQL commands that this query builds
  show_query(query)

  print('-------------------------')
  
  # Actually get the Boxscores data -- using collect against the query
  results_df <- query |>
    collect()
  
  # Disconnect
  dbDisconnect(con)
  
  print(results_df)
  print("Done")
  
}, error = function(e) {
  cat("Error occurred:\n")
  cat(sprintf("  Message: %s\n", e$message))
  cat(sprintf("  Attempted path: %s\n", path_to_database))
  
  # Ensure connection is closed if it was opened
  if (exists("con") && dbIsValid(con)) {
    dbDisconnect(con)
  }
})